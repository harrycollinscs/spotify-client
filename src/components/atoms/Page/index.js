import React from "react";
import styled from "styled-components";
import NavigationBar from "../../organisms/NavBar";

const StyledDiv = styled.div`
  overflow-y: scroll;
  margin-top: ${(props) => (props.withNavigation ? '8vh' : '0')};
  height: 100vh;
  width; 85vw;
  padding: 0px 0px 50px 0px;
  background-color: ${(props) => (props.dark ? "black" : "white")};
  position: relative;
`;

const navigationLinks = [
  {
    title: "Songs",
    route: "/songs",
  },
  {
    title: "Artists",
    route: "/artists",
  },
  {
    title: "Playlists",
    route: "/playlists",
  },
];

const Page = ({ children, withNavigation = true, darkBackground = false }) => (
  <StyledDiv dark={darkBackground} withNavigation={withNavigation}>
    {withNavigation && <NavigationBar links={navigationLinks} />}
    {children}
  </StyledDiv>
);

export default Page;
