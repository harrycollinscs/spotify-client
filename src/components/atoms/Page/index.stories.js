import React from "react";
import Page from "./index";

export default {
  title: "Atoms/Page",
  component: Page,
  argTypes: {
    withNavigation: {
      options: [true, false],
      control: { type: "radio" },
    },
    darkBackground: {
      options: [true, false],
      control: { type: "radio" },
    },
  },
};

const Template = (args) => <Page {...args} />;

export const Default = Template.bind({});

export const WithoutNavigation = Template.bind({});
WithoutNavigation.args = {
  withNavigation: false,
  children: "Child of the page",
};

export const WithDarkBackground = Template.bind({});
WithDarkBackground.args = {
  darkBackground: true,
  children: "Child of the page",
};
