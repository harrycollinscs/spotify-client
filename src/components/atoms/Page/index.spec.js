import React from "react";
import Page from "./index";
import { shallow } from "enzyme";

describe("Atoms/Page", () => {
  const component = <Page />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
