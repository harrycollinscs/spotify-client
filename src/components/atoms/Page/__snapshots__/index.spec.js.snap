// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Page Should match snapshot 1`] = `
<styled.div
  dark={false}
  withNavigation={true}
>
  <NavigationBar
    links={
      Array [
        Object {
          "route": "/songs",
          "title": "Songs",
        },
        Object {
          "route": "/artists",
          "title": "Artists",
        },
        Object {
          "route": "/playlists",
          "title": "Playlists",
        },
      ]
    }
  />
</styled.div>
`;
